tests/Integration/PrestaShopBundle/Translation/Provider/TranslationsCatalogueProviderTest.php

```diff
@@ -0,0 +1,324 @@
+            ],
+            $messages['ModulesCheckpaymentShop']['Pay by check']
+        );
+    }
+
+    /**
+     * The provider should retrieve all translations from files that
+     * look like `AdminSomething` or `ModulesSomethingAdmin`
+     */
+    public function testGetDomainCatalogueForModule()
+    {
+        $providerFactory = $this->createMock(ProviderFactory::class);
+
+        $providerFactory
+            ->expects($this->any())
+            ->method('build')
+            ->willReturn(
+                new ModulesProvider(
+                    $this->databaseReader,
+                    $this->getBuiltInModuleDirectory(),
+                    $this->getDefaultTranslationsDirectory(),
+                    $this->container->get('prestashop.translation.loader.legacy_file'),
+                    $this->container->get('prestashop.translation.extractor.legacy_module'),
+                    'checkpayment'
+                )
+            );
+
+        $provider = new TranslationsCatalogueProvider($providerFactory);
+        $messages = $provider->getDomainCatalogue(
+            new ModulesType('checkPaymentShop'),
+            'fr-FR',
+            'ModulesCheckpaymentShop'
+        );
+        $this->assertIsArray($messages);
+
+        // Check integrity of translations
+        $this->assertCount(19, $messages);
+        $this->assertArrayHasKey('default', $messages[0]);
+        $this->assertArrayHasKey('xliff', $messages[0]);
+        $this->assertArrayHasKey('database', $messages[0]);
+        $this->assertArrayHasKey('tree_domain', $messages[0]);
+        $this->assertSame([
+            'Modules',
+            'Checkpayment',
+            'Shop',
+        ], $messages[0]['tree_domain']);
+
+        // search a word
+        $messages = $provider->getDomainCatalogue(
+            new ModulesType('checkPaymentShop'),
+            'fr-FR',
+            'ModulesCheckpaymentShop',
+            ['payment']
+        );
+        $this->assertIsArray($messages);
+
+        // Check integrity of translations
+        $this->assertCount(3, $messages);
+        $this->assertArrayHasKey('default', $messages[0]);
+        $this->assertArrayHasKey('xliff', $messages[0]);
+        $this->assertArrayHasKey('database', $messages[0]);
+        $this->assertArrayHasKey('tree_domain', $messages[0]);
+        $this->assertSame([
+            'Modules',
+            'Checkpayment',
+            'Shop',
+        ], $messages[0]['tree_domain']);
+    }
+
+    protected function tearDown()
+    {
+        $langId = Language::getIdByIso('fr', true);
+        if ($langId) {
+            \Db::getInstance()->execute(
+                'DELETE FROM `' . _DB_PREFIX_ . 'lang` WHERE id_lang = ' . $langId
+            );
+        }
+        self::$kernel->shutdown();
+    }
+
+    /**
+     * @return string
+     */
+    private function getBuiltInModuleDirectory(): string
+    {
+        return __DIR__ . '/../../../../Resources/modules';
```

Comment:
Maybe we could improve this with an ENV variable ?

If one day we decide to move this tests, this will crash, and I'm worried about `../../../..` things
At least check directory content 😅  ?

Reply:
It must be defined in the test environment like we do with `%kernel.root_dir%`